### PR TITLE
Added compressors and decoders

### PR DESCRIPTION
**Pull request by scohen**
The client now supports these encoders:

  1. erlpack
  2. Json
  3. MessagePack

And gzip compression.

Also went through and added docs.

### CODE REVIEW

File: ex/loqui/lib/loqui/client.ex [hunk 1]
```diff
@@ -1,5 +1,89 @@
...
+  # A high performance client for the Loqui protocol
+
+  This client exposes five public functions, `start_link`, `request`, `push`, `ping` and `close`.
+  In loqui parlance, `request` is a synchronous call and `push` is asynchrounous. Be careful with push
+  calls, as they can overwhelm the remote server.
+
+  ## Codecs and Compressors
+
+  A `codec` changes data structures into `iodata` and a `compressor` applies a compression algorithm
+  to iodata. They're applided thusly:
```
Comment (jhowarth) [inline ex/loqui/lib/loqui/client.ex]: [nit] `s/applided/applied`

File: ex/loqui/lib/loqui/client.ex [hunk 2]
```diff
@@ -1,5 +1,89 @@
 defmodule Loqui.Client do
+  @moduledoc """
+  # A high performance client for the Loqui protocol
+
+  This client exposes five public functions, `start_link`, `request`, `push`, `ping` and `close`.
```
Comment (jhowarth) [inline ex/loqui/lib/loqui/client.ex]: Why is ping exposed? Shouldn't it just be handled automatically?

File: ex/loqui/lib/loqui/client.ex [hunk 3]
```diff
@@ -76,8 +160,8 @@ defmodule Loqui.Client do
 
   @default_timeout 5000
   @max_sequence round(:math.pow(2, 32) - 1)
-  @default_compressors %{Compressors.NoOp.name() => Compressors.NoOp}
-  @default_codecs %{Codecs.Erlpack.name() => Codecs.Erlpack}
+  @default_compressors Enum.into(Compressors.all(), %{}, &{&1.name, &1})
```
Comment (jhowarth) [inline ex/loqui/lib/loqui/client.ex]: Interesting that this is `.name` but was `.name()` in the previous code. Is it something about anonymous functions and macros?

File: ex/loqui/lib/loqui/client.ex (hunk 2 above)
Comment (scohen) [inline ex/loqui/lib/loqui/client.ex]: This is if you want to manually ping the server for some reason. I can remove it, it's not essential.

File: ex/loqui/lib/loqui/client.ex (hunk 3 above)
Comment (scohen) [inline ex/loqui/lib/loqui/client.ex]: Nope, just me being inconsistent. `.name()` and `.name` are the same.

File: ex/loqui/lib/loqui/client.ex (hunk 1 above)
Comment (scohen) [inline ex/loqui/lib/loqui/client.ex]: ✅

File: ex/loqui/lib/loqui/client.ex (hunk 3 above)
Comment (scohen) [inline ex/loqui/lib/loqui/client.ex]: ✅

File: ex/loqui/lib/loqui/client.ex (hunk 2 above)
Comment (scohen) [inline ex/loqui/lib/loqui/client.ex]: Sometimes, having a ping function is a good sanity check that you got things working at a low level.

**Pull request merged**